spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/jdbc/TestDatabaseAutoConfigurationTests.java

```diff
@@ -60,6 +63,41 @@ void replaceWithUniqueDatabase() {
 		});
 	}
 
+	@Test
+	void testRunningSchemaSqlScriptEvenWhenSchemaCredentialsAreSet() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(DataSourceAutoConfiguration.class))
+				.withPropertyValues("spring.datasource.schema-username=sa",
+						"spring.datasource.schema=" + getRelativeLocationFor("schema.sql"),
+						"spring.datasource.data=" + getRelativeLocationFor("data.sql"))
+				.run((context) -> {
+					DataSource dataSource = context.getBean(DataSource.class);
+					JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+					assertNumberOfRows(jdbcTemplate, 1);
+				});
+	}
+
+	@Test
+	void testRunningSchemaSqlScriptEvenWithWrongCredentials() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(DataSourceAutoConfiguration.class))
+				.withPropertyValues("spring.datasource.schema-username=sa", "spring.datasource.schema-password=sa",
+						"spring.datasource.schema=" + getRelativeLocationFor("schema.sql"),
+						"spring.datasource.data=" + getRelativeLocationFor("data.sql"))
+				.run((context) -> {
+					assertThat(context).hasFailed();
+					assertThat(context).getFailure()
+							.hasMessageContaining("Failed to obtain JDBC Connection; nested exception is")
+							.hasMessageContaining("Wrong user name or password");
```

Comment:
That test looks wrong to me. If a test setup has replaced the production database by an embedded one, then the credentials of the production database should not impact the ability to initialize the embedded database.

These  settings have been introduced to perform the initialization of a "production" database with different credentials (and only for that step). That shouldn't apply to an embedded database IMO.

Reply:
Yes, you are right.